go-controller/pkg/util/node_annotations.go

```diff
@@ -309,3 +309,17 @@ func ParseNodeHostAddresses(node *kapi.Node) (sets.String, error) {
 
 	return sets.NewString(cfg...), nil
 }
+
+// GetK8sNodeAddress returns the K8s Node's Cluster internal IP address
+func GetK8sNodeAddress(node *kapi.Node) (net.IP, error) {
```

Comment:
Will the internal IP address be populated on node status before this function gets executed when scaling a node?

Reply:
When scaling the cluster, a node first joins the cluster and only then workloads are scheduled on it.
So I expect this bit to run after the the IP has populated.
If the Address is missing then ovnkube node will fail so and the pod will get recreated.

Looking into the Node code further, `node.go:setupOVNNode()` takes the same approach.
That said, `util.GetNodePrimaryIP()` is awfully similar to this :) So i will re-use it.